From b97c51d6a2ca70a5e60209cd7ab16563455b8d13 Mon Sep 17 00:00:00 2001
From: Dominik Csapak <d.csapak@proxmox.com>
Date: Wed, 14 Dec 2016 08:40:02 +0100
Subject: [PATCH 8/9] focus canvas after load

or else in some browsers, the canvas does not get the focus after
loading in an iframe

Signed-off-by: Dominik Csapak <d.csapak@proxmox.com>
---
 app/ui.js | 2 ++
 vnc.html  | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/app/ui.js b/app/ui.js
index d32c789..ed2897b 100644
--- a/app/ui.js
+++ b/app/ui.js
@@ -442,6 +442,8 @@ const UI = {
                     msg = _("Connected (unencrypted) to ") + UI.desktopName;
                 }
                 UI.showStatus(msg);
+		rfb.get_keyboard().set_focused(true);
+		document.getElementById("noVNC_canvas").focus();
                 break;
             case 'disconnecting':
                 UI.connected = false;
diff --git a/vnc.html b/vnc.html
index 1423f78..ff70e0c 100644
--- a/vnc.html
+++ b/vnc.html
@@ -332,7 +332,7 @@
                 autocorrect="off" autocomplete="off" spellcheck="false"
                 mozactionhint="Enter"></textarea>
 
-            <canvas id="noVNC_canvas" width="0" height="0">
+            <canvas id="noVNC_canvas" width="0" height="0" tabindex="0">
                         Canvas not supported.
             </canvas>
         </div>
-- 
2.11.0

